expo/expo-cli
# feat: Prompt user for git init if in existing git work tree

# Why

This PR addresses Issue #2378.

When using `expo init`, the expo-cli automatically runs `git init` inside the project folder. However, if the user is running the command line tool within an existing Git Working Tree, then `git init` has the effect of creating a submodule. This is generally undesired behaviour. In the situation of running the command line tool within an existing Git Working Tree, we should give the user the option to avoid running `git init`.

# How

Checking whether or not the user was in an existing Git Working tree was actually done already. I just needed to use the result of that check to bring up a prompt to the user. The prompt asks 'y/n' and appropriately runs `git init` or not based on the answer.

# Test Plan

The tests were very manual. To reproduce

- Test on existing git working tree
  1. Create an empty folder, and run `git init`.
  2. Within that folder, run `expo init`
  3. It should prompt you whether or not to run 'git init'.
  4. Check the resulting folder for a `.git` folder.
- Test on fresh folder 
  1. Create an empty folder that is not within a git repository
  2. Within that folder, run `expo init`
  3. It should NOT prompt you whether or not to run 'git init'.
  4. Check the resulting folder for a `.git` folder. It should be there.
